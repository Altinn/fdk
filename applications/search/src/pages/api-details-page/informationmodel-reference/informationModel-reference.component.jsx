import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Link } from 'react-router-dom';
import { getTranslateText } from '../../../lib/translateText';
import localization from '../../../lib/localization';

export const InformationModelReference = props => {
  const { informationModelReference } = props;
  if (!informationModelReference) {
    return null;
  }

  const id = _.get(informationModelReference, ['id']);
  const prefLabel = _.get(informationModelReference, ['title']);

  return (
    <div className="d-flex flex-column list-regular--item mb-4">
      <Link to={`/informationModels/${id}`}>
        {localization.api.linkInformationModelReference}
      </Link>
      <strong>{prefLabel ? getTranslateText(prefLabel) : id}</strong>
    </div>
  );
};

InformationModelReference.defaultProps = {
  informationModelReference: null
};

InformationModelReference.propTypes = {
  informationModelReference: PropTypes.object
};
