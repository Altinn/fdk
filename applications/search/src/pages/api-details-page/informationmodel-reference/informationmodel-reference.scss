.informationmodel-item {
  padding-bottom: 1px;
}
