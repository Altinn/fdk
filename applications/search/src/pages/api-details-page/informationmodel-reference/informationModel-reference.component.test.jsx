import React from 'react';
import { shallow } from 'enzyme';
import { InformationModelReference } from './informationmodel-reference.component';
import informationModelItem from '../../../mock/informationmodels.response.json';

let defaultProps;

test('should render InformationModelReference correctly with no props', () => {
  const wrapper = shallow(<InformationModelReference />);
  expect(wrapper).toMatchSnapshot();
});

test('should render InformationModelReference correctly', () => {
  defaultProps = {
    informationModelReference: informationModelItem,
    index: 0
  };
  const wrapper = shallow(<InformationModelReference {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
