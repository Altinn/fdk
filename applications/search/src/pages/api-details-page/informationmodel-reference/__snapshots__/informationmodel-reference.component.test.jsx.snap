// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InformationModelReference correctly 1`] = `
<div
  className="d-flex flex-column list-regular--item mb-4"
>
  <Link
    replace={false}
    to="/informationModels/936c581f-9d95-418b-8fce-e452f6bdfd5e"
  >
    <strong>
      Ã…pne Data fra Enhetsregisteret - API Dokumentasjon
    </strong>
  </Link>
</div>
`;

exports[`should render InformationModelReference correctly with no props 1`] = `""`;
